contracts/token-registrars/StandardizedTokenRegistrar.sol

```diff
@@ -0,0 +1,159 @@
+    }
+
+    /**
+     * @notice Getter for the contract id.
+     */
+    function contractId() external pure returns (bytes32) {
+        return CONTRACT_ID;
+    }
+
+    function getStandardizedTokenSalt(address deployer, bytes32 salt) public view returns (bytes32) {
+        return keccak256(abi.encode(PREFIX_STANDARDIZED_TOKEN_SALT, chainNameHash, deployer, salt));
+    }
+
+    function getStandardizedTokenId(address deployer, bytes32 salt) public view returns (bytes32 tokenId) {
+        tokenId = service.getCustomTokenId(address(this), getStandardizedTokenSalt(deployer, salt));
+    }
+
+    function getStandardizedTokenAddress(address deployer, bytes32 salt) public view returns (address tokenAddress) {
+        tokenAddress = service.getStandardizedTokenAddress(getStandardizedTokenId(deployer, salt));
+    }
+
+    function deployStandardizedToken(
+        bytes32 salt,
+        string calldata name,
+        string calldata symbol,
+        uint8 decimals,
+        uint256 mintAmount,
+        address distributor
+    ) external payable {
+        address sender = msg.sender;
+        salt = getStandardizedTokenSalt(sender, salt);
+        bytes32 tokenId = service.getCustomTokenId(address(this), salt);
+
+        service.deployAndRegisterStandardizedToken(salt, name, symbol, decimals, mintAmount, distributor);
+        ITokenManager tokenManager = ITokenManager(service.getTokenManagerAddress(tokenId));
+        tokenManager.transferOperatorship(sender);
+
+        IStandardizedToken token = IStandardizedToken(service.getStandardizedTokenAddress(tokenId));
+        token.safeTransfer(sender, mintAmount);
+    }
+
+    function deployRemoteStandarizedToken(
+        bytes32 salt,
+        address additionalDistributor,
+        address optionalOperator,
+        uint256 mintAmount,
+        string memory destinationChain,
+        uint256 gasValue
+    ) external payable {
+        string memory tokenName;
+        string memory tokenSymbol;
+        uint8 tokenDecimals;
+        bytes memory distributor = new bytes(0);
+        bytes memory operator = new bytes(0);
+
+        {
+            address sender = msg.sender;
+            salt = getStandardizedTokenSalt(sender, salt);
+            bytes32 tokenId = service.getCustomTokenId(address(this), salt);
+
+            IStandardizedToken token = IStandardizedToken(service.getStandardizedTokenAddress(tokenId));
+            ITokenManager tokenManager = ITokenManager(service.getTokenManagerAddress(tokenId));
+
+            tokenName = token.name();
+            tokenSymbol = token.symbol();
+            tokenDecimals = token.decimals();
+            if (additionalDistributor != address(0)) {
+                if (!token.isDistributor(additionalDistributor)) revert NotDistributor(additionalDistributor);
+                distributor = additionalDistributor.toBytes();
+            } else if (mintAmount != 0) {
+                revert NonZeroMintAmount();
```

Comment:
add a comment that if distributor is not set than mintAmount must be 0 to enforce that additional token deployments can't change it